nobl9/resource_agent.go

```diff
@@ -106,11 +116,18 @@ func resourceAgentApply(ctx context.Context, d *schema.ResourceData, meta interf
 	var p n9api.Payload
 	p.AddObject(service)
 
-	err := client.ApplyObjects(p.GetObjects())
+	agentsData, err := client.ApplyAgents(p.GetObjects())
 	if err != nil {
 		return diag.Errorf("could not add agent: %s", err.Error())
 	}
 
+	if agentsData != nil {
```

Comment:
Could you also check if there is only one element in the response? Instead of using this magic `agentsData[len(agentsData)-1]`, you could retrieve just the first element like `agentsData[0]` - what do you think?

Reply:
sure, I think it's a good idea. THX